src/protocol/primitives/record-batch/record-batch.ts

```diff
@@ -81,7 +113,6 @@ export class RecordBatch implements Serializable {
     const expectedCrc32 = Int32.deserialize(temporary);
     const calculatedCrc32c = crc32c(temporary.toBuffer(temporary.getOffset()));
     if (calculatedCrc32c !== expectedCrc32.value) {
```

Comment:
```suggestion
    if (calculatedCrc32c !== expectedCrc32.value) {
      throw new InvalidRecordError(`Record is corrupt (stored crc = ${expectedCrc32.value}, computed crc = ${calculatedCrc32c})`);
```

https://github.com/apache/kafka/blob/647afeff6a2e3fd78328f6989e8d9f96bcde5121/clients/src/main/java/org/apache/kafka/common/record/DefaultRecordBatch.java#L130C23-L130C45